pkgs/build-support/rust/import-cargo-lock.nix

```diff
@@ -79,17 +81,16 @@ let
   # We can't use the existing fetchCrate function, since it uses a
   # recursive hash of the unpacked crate.
   fetchCrate = pkg:
-    assert lib.assertMsg (pkg ? checksum) ''
+    let
+      checksum = pkg.checksum or parsedLockFile.metadata."checksum ${pkg.name} ${pkg.version} (${pkg.source})";
+    in
+    assert lib.assertMsg (checksum != null) ''
       Package ${pkg.name} does not have a checksum.
-      Please note that the Cargo.lock format where checksums used to be listed
-      under [metadata] is not supported.
-      If that is the case, running `cargo update` with a recent toolchain will
-      automatically update the format along with the crate's depenendencies.
     '';
```

Comment:
Maybe I should adjust the message to say something like "the lock file may be corrupted, you can generate one with ...", instead of now just leaving it vague?

Reply:
This also does beg the question of why v1 support was intentionally omitted from the initial implementation -- is there some issue we may run into? I haven't run into anything in my limited testing, at least, so I hope that's not it.

Reply:
Leaving it as it is lgtm. I think suggesting running `cargo update` could be helpful, but I don't think we should say it might be corrupted, since the issue is probably more likely to be `impoCargoLock` instead of  `Cargo.lock`